iceoryx_utils/test/moduletests/test_posix_timer.cpp

```diff
@@ -520,3 +520,24 @@ TIMING_TEST_F(Timer_test, CatchUpPolicySkipToNextBeatCallsLessCallbacksThanASAPT
 
     TIMING_TEST_EXPECT_TRUE(softTimerCounter < asapTimerCounter);
 });
+
+TEST_F(Timer_test, DISABLED_SelfTriggeringTimerWorksAndDoesNotCauseSegFault)
+{
+    Duration selfTriggerTimeout = 1_ns;
+    int repetitions = 100;
+    std::atomic_int counter{0};
+    {
+        Timer sut(selfTriggerTimeout, [&] {
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+            if (counter < repetitions)
+            {
+                EXPECT_FALSE(
+                    sut.restart(selfTriggerTimeout, Timer::RunMode::ONCE, Timer::CatchUpPolicy::IMMEDIATE).has_error());
+            }
+            ++counter;
+        });
+
+        sut.start(Timer::RunMode::ONCE, Timer::CatchUpPolicy::IMMEDIATE);
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
If this sleep expires, the timer may still be active if I did not miss something. Is this intended? The sut dtor will be called, the counter go out of scope, the callback could access data now that is gone (either from the timer, but in any case the counter itself). There is no other join or wait mechanism that waits until all repetitions (or even one) were performed.

Conclusion: I may not fully understand the test but I think object lifetime is a problem here.

Reply:
From my point of view this tests is valid code and it should not segfault. We just wait here 1000 ms since it is sufficient time to provoke the segfault.
I share the same conclusion but at the moment I do not have the time to solve this segfault, therefore I created #243 .

Reply:
AFAIK you cannot stop a POSIX timer reliably once the kernel has decided to start the callback (but is not yet running). I encountered this issue while trying to implement a timed_wait function in another project. And if you cannot stop it, you have to make sure that the data the callback tries to access is still there (even if it is just a flag/counter).

Hard for me to say whether the timer is broken, but I guess what you are saying is that the test is not responsible to block (a bit dangerous in case of a malfunction of the timer but maybe unavoidable).

TLDR: I suspect there is a fundamental flaw in the timer in the small window before the callback actually runs.